hail/python/hail/expr/aggregators/aggregators.py

```diff
@@ -1508,13 +1508,52 @@ def linreg(y, x, nested_dim=1, weight=None) -> StructExpression:
 
     hl.methods.statgen._warn_if_no_intercept('linreg', x)
 
-    if weight is None:
-        return _linreg(y, x, nested_dim)
-    else:
-        return _linreg(hl.sqrt(weight) * y,
-                       [hl.sqrt(weight) * xi for xi in x],
-                       nested_dim)
+    if weight is not None:
+        y = hl.sqrt(weight) * y
+        x = [hl.sqrt(weight) * xi for xi in x]
+    temp = _linreg(y, x, nested_dim)
 
+    k = len(x)
+    k0 = nested_dim
+    n = hl.agg.count_where(hl.all(lambda x: hl.is_defined(x), x) & hl.is_defined(y))
+    yty = hl.agg.filter(hl.is_defined(x), hl.agg.sum(y*y))
+    xty = temp.xty
+    beta = temp.beta
+    diag_inv = temp.diag_inv
+    beta0 = temp.beta0
+
+    def dot(a, b):
+        return hl.sum(a * b)
+
+    d = n - k
```

Comment:
I'm a bit worried about the rest of the code here getting inlined for each linreg -- we're trying to make it possible for Konrad to run 2500 linregs as part of a table aggregate, and I imagine this will blow method size limits.

Let's register a function as we do above in `hist`?

Reply:
(the function should take as arguments the `_linreg` struct and the `n, yty` struct)